### ISSUE TITLE
Multi arch build?

### ISSUE THREAD

**Issue by dmarkey** [kubernetes/git-sync#222]
Can we get multi arch builds by any chance? I needed to build my own image for a Raspberry Pi. Kubernetes on arm is getting more popular.

**Comment by thockin** [comment]
We have been building a multi-arch manifest lest for a while.  For example:

https://console.cloud.google.com/gcr/images/google-containers/GLOBAL/git-sync@sha256:fbc34d1294951a079ff6cbef9a15e870200cbf703cb57d3c817303778596e810/details?tab=info


```
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1159,
         "digest": "sha256:9d78213e442d0fc9a08a3e8d97f2e3992c45b10532d557302e5fb69362c39360",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1159,
         "digest": "sha256:24450aa549f9cf954a8c19e75b3c3030075ab40de5b8249b381f70bd2c063fe6",
         "platform": {
            "architecture": "arm",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1159,
         "digest": "sha256:31724b1ce6ebfddaa950db96e9f2be1f8a75be26bba94bfcabc83850661fd8aa",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1159,
         "digest": "sha256:6136c12e7239e81c38032556395a0617ef00f5dca424df429c55a2be56e4c8f4",
         "platform": {
            "architecture": "ppc64le",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1159,
         "digest": "sha256:aff4ef19c9c782ac6c86ead3de6e4621918199be62f5909cae5157c688bdfddf",
         "platform": {
            "architecture": "s390x",
            "os": "linux"
         }
      }
   ]
}
```

Does that not work for you?

**Comment by dmarkey** [comment]
Oh let me try again. I thought got an exec format error on my Raspberry pi.

**Comment by thockin** [comment]
It's possible that I messed something up - let me know :)

**Comment by dmarkey** [comment]
Oh, I was trying with the hugo demo deployment, it was using an old image.

Here's a PR to use the latest which works on my RaspberryPi

https://github.com/kubernetes/git-sync/pull/223/files

**Comment by flavio** [comment]
I fear something is still broken, I'm still getting the `exec format error` both with the `k8s.gcr.io/git-sync:v3.1.6` and the `v3.1.3` one (the one referenced by @dmarkey above).

This is happening on my ARM64 nodes.

**Comment by flavio** [comment]
More details: the `git-sync` binary shipped with the image has the correct architecture. However, all the other binaries (including git) have the x86_64 architecture.

ping @thockin

**Comment by thockin** [comment]
Indeed.  Looking at it

**Comment by thockin** [comment]
Right, so I don't know how I ever thought this would work.  The build pulls an x86 base image, runs some apt-get in it, and copies an arch-specific binary in. That's not going to work.

I'm not sure what the right fix for this is.  I apologize for the confusion.

**Comment by thockin** [comment]
OK, I am exploring a buildx solution.  I'll push a test image somewhere and maybe you can try it for me, to confirm?

**Comment by thockin** [comment]
Can you try with `thockin/git-sync:v3.1.6-dirty__linux_arm64` and/or `thockin/git-sync:v3.1.6-dirty`

**Comment by flavio** [comment]
@thockin sorry for the late response. I can confirm the `thockin/git-sync:v3.1.6-dirty` works as expected both on x86_64 and on ARM64
